File: apps/api/Models/Entities/Workspace.cs
namespace Align.Models;

public class Workspace : Entity
{
    public required string Name { get; set; }
    public required string Url { get; set; }
    public string? LogoUrl { get; set; }

    public ICollection<User> Users { get; set; } = [];
    public ICollection<WorkspaceMember> WorkspaceMembers { get; set; } = [];
}

File: packages/types-api/src/workspace.ts
import { Guid } from '@align/utility-types'

import { User } from './user'

export type Workspace = {
  id: Guid
  name: string
  url: string
  logoUrl?: string
}

export type CreateWorkspacePayload = {
  name: string
  url: string
  companySize: string
  role: string
}

export type WorkspaceValidationError = 'UrlTaken' | 'InvalidUrl'

export type CreateWorkspaceResult =
  | {
      isSuccess: true
      workspace: Workspace
      user?: User
    }
  | {
      isSuccess: false
      validationError: WorkspaceValidationError
    }

export type WorkspaceMemberRole = 'Guest' | 'Member' | 'Admin'
